tgdwyer/WebCola
# Add ability to change arrow head length for edge routing

This is done by adding `ah` parameter to the method `Layout.routeEdge` in `layout.ts` file. However changing the arrow head length this way is not ideal, since there is an optional `draw` parameter in between the two parameters.

For example, if we want to change the arrow head length from 5 (which is set to be the default value for backwards compatibility) to 15, we would do the following:
```
// Modify this line...
cola.routeEdge(d);
// ...so that the line looks like this
cola.routeEdge(d, undefined, 15);
```

## Potential issues

Whilst running `grunt` (after installing all dependencies), I got a timeout error from the qunit task. All other tasks appears to have worked correctly. 
```
$ grunt
(node:23560) ExperimentalWarning: The http2 module is an experimental API.
Running "copy:qunit" (copy) task
Copied 2 files

Running "ts:commonjs" (ts) task
Compiling...
Cleared fast compile cache for target: commonjs
Using tsc v2.6.2
>> Warning: created WebCola/.baseDir.ts file because `outDir` was specified in the tsconfig.json, but not `rootDir`.Add `rootDir`  in the `compilerOptions` section to fix this warning.



TypeScript compilation complete: 3.61s for 2 TypeScript files.

Running "ts:test" (ts) task
Compiling...
Cleared fast compile cache for target: test
Using tsc v2.6.2



TypeScript compilation complete: 3.38s for 2 TypeScript files.

Running "browserify:dist" (browserify) task
>> Bundle WebCola/cola.js created.

Running "browserify:examples" (browserify) task
>> Bundle WebCola/examples/tmdbgraph.js created.
>> Bundle WebCola/examples/3dtree.js created.
>> Bundle WebCola/examples/statemachinepowergraph.js created.
>> Bundle WebCola/examples/pretrip.js created.
>> Bundle WebCola/examples/powergraphexample.js created.
>> Bundle WebCola/examples/dotpowergraph.js created.
>> Bundle WebCola/examples/3dlayout.js created.
>> Bundle WebCola/examples/vhybridize.js created.
>> Bundle WebCola/examples/tetrisbug.js created.

Running "browserify:test" (browserify) task
>> Bundle WebCola/test/bundle.js created.

Running "uglify:dist" (uglify) task
>> 1 sourcemap created.
>> 1 file created 416.81 kB → 79.48 kB

Running "qunit:all" (qunit) task
Testing WebCola/test/test.html .......
>> PhantomJS timed out, possibly due to:
>> - QUnit is not loaded correctly.
>> - A missing QUnit start() call.
>> - Or, a misconfiguration of this task.
>> 1 tests completed with 1 failed, 0 skipped, and 0 todo.
>> 0 assertions (in 0ms), passed: 0, failed: 0

Done.
```

## Manual testing
I have manually tested the WebCola library against the "Unix Family Tree" example, to see if changing the arrow head length would reflect that on the browser. It works as expected.

### Arrow head length set to default (5)
`return lineFunction(d3cola.routeEdge(d));`
![Arrow head length is set to default, which is 5](https://user-images.githubusercontent.com/5346136/34802689-0f9ce8fe-f6c3-11e7-9d3f-c8b3232d2c48.png)

### Arrow head length set to 0
`return lineFunction(d3cola.routeEdge(d, undefined, 0));`
![Arrow head length is set to 0](https://user-images.githubusercontent.com/5346136/34802713-34501acc-f6c3-11e7-9217-49024bd83477.png)

### Arrow head length set to 50
`return lineFunction(d3cola.routeEdge(d, undefined, 50));`
![Arrow head length is set to 50](https://user-images.githubusercontent.com/5346136/34802722-41ab419c-f6c3-11e7-9702-a376a636aacb.png)
